part/map.go

```diff
@@ -167,6 +226,9 @@ func (m Map[K, V]) SlowEqual(other Map[K, V]) bool {
 		return true
```

Comment:
I think this case is broken for two singletons with differing contents. Needs the same logic as above. (And possibly a test which tests SlowEqual for this case)

Reply:
Oh indeed, forgot to update that. Fixed and added a quick test to validate the properties.